soc/arm/renesas_smartbond/da1469x/soc.c

```diff
@@ -31,13 +35,53 @@
 static uint32_t z_renesas_cache_configured;
 #endif
 
+#if defined(CONFIG_PM)
+struct dcdc_regs {
+	uint32_t v18;
+	uint32_t v18p;
+	uint32_t vdd;
+	uint32_t v14;
+	uint32_t ctrl1;
+};
+
+static struct dcdc_regs dcdc_state;
+#endif
+
 void sys_arch_reboot(int type)
 {
 	ARG_UNUSED(type);
 
 	NVIC_SystemReset();
 }
 
+void z_smartbond_restore_dcdc_state(void)
+{
+	/*
+	 * Enabling DCDC while VBAT is below 2.5 V renders system unstable even
+	 * if VBUS is available. Enable DCDC only if VBAT is above minimal value.
+	 */
+	if (CRG_TOP->ANA_STATUS_REG & CRG_TOP_ANA_STATUS_REG_COMP_VBAT_HIGH_Msk) {
+#if defined(CONFIG_PM)
+		DCDC->DCDC_V18_REG = dcdc_state.v18;
+		DCDC->DCDC_V18P_REG = dcdc_state.v18p;
+		DCDC->DCDC_VDD_REG = dcdc_state.vdd;
+		DCDC->DCDC_V14_REG = dcdc_state.v14;
+		DCDC->DCDC_CTRL1_REG = dcdc_state.ctrl1;
+#endif
+	}
+}
+
+void z_smartbond_store_dcdc_state(void)
+{
+#if defined(CONFIG_PM)
+	dcdc_state.v18 = DCDC->DCDC_V18_REG;
+	dcdc_state.v18p = DCDC->DCDC_V18P_REG;
+	dcdc_state.vdd = DCDC->DCDC_VDD_REG;
+	dcdc_state.v14 = DCDC->DCDC_V14_REG;
+	dcdc_state.ctrl1 = DCDC->DCDC_CTRL1_REG;
+#endif
+}
```

Comment:
why is this touching DC/DC without regulator API?

Reply:
When device goes to sleep some register values (including DC/DC) are lost. This code stores settings on sleep and restores them on wakeup. Foe each rail that do have DCDC option only DCDC register values are lost. Other configuration (i.e. voltages) are kept intact. Sleep in Smartbond also looses CPU registers that are restored in PM code. To make things fast there is no call to regulator API since no change in regulator is expected.

Reply:
this should be handled by device drivers, not in a random file that does not even respect DT (what is "DCDC"?)

Reply:
This has nothing to do with DT or device driver. Extended sleep on DA1469x effectively powers off ARM core and we need to store some registers to restore them quickly on wakeup. If we pass this through all the subsystems in Zephyr instead of just restoring from local variables, the wake up process will be unnecessarily long for no reason.

Reply:
because we only need to do this in case ARM core and was shut down, otherwise those registers are not affected - PM events do not know if that was the case